app/bin/tools/backfill_likes.dart

```diff
@@ -31,8 +34,14 @@ Future main(List<String> args) async {
 
     useLoggingPackageAdaptor();
     await for (Like l in dbService.query<Like>().run()) {
-      final f = pool.withResource(() => _backfillLikes(l));
-      futures.add(f);
+      final r = await pool.request();
+      futures.add(() async {
```

Comment:
I find this really confusing, and it took my a while to interpret what this does and to confirm that it does the same (with more lines). What was your issue with the previous approach?

Reply:
Well, I still have an unbounded list of futures... but I don't schedule an unbounded list of micro-tasks..

In general, we have a problem in these backfill utilities.. we should always do operations streaming. And block reading the query over all `Like` entities.

Reply:
Please add a comment that describes this. `package:executor` has a nice construct to check on the scheduled but not yet running items, and we can selectively wait on the currently running and all of the scheduled ones. In the future we may use that instead of `package:pool`.

Reply:
The `Executor.onChange` could probably be used to do this.. but it also seems a bit complex...

Ideally, we need a function to concurrently map a stream with a concurrency limit.

Reply:
Yeah, a stream transform/processing utility with a concurrency would be nice...